### PR TITLE
Update  Azure SQLDB 2020-02-02-preview swagger files

### PR DESCRIPTION
**Pull request by ericshape**
Update Azure SQLDB 2020-02-02-preview swagger files. 

Most of the changes (>99%) in this PR are **auto-generated** by SQL DB DSMainDev Repo. The manual change is only in README.md and readme.azureresourceschema.md.

This PR is a copy of an existing API-version to a preview version (2020-02-02-preview) with no API changes. Via this PR, we will address over **[73 S360 Alerts](https://vnext.s360.msftcloudes.com/blades/unifiedplatform?global=4:a7a7c4d7-03e5-4816-add0-30646ccee978&blade=KPI:6dc4e485-2d1b-449c-a224-30b183391437~SLA:3~DueDate:~AssignedTo:All~Forums:All~_loc:UnifiedPlatform&tile=)** about API swagger completeness and correctness.

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [ ] update existing version for new feature 
      - [x] update existing version to fix swagger quality issue in s360:  **[73 S360 Alerts Link](https://vnext.s360.msftcloudes.com/blades/unifiedplatform?global=4:a7a7c4d7-03e5-4816-add0-30646ccee978&blade=KPI:6dc4e485-2d1b-449c-a224-30b183391437~SLA:3~DueDate:~AssignedTo:All~Forums:All~_loc:UnifiedPlatform&tile=)**
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
_Already published_
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
_Already published_
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [x] SDK of .NET (need service team to ensure code readiness)
      - [x] SDK of Python
      - [x] SDK of Java
      - [x] SDK of Js
      - [x] SDK of Go
      - [x] PowerShell
      - [x] CLI
      - [x] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

**The breaking change errors are not real errors.** 
**In https://github.com/Azure/azure-rest-api-specs/pull/11840/checks?check_run_id=1465548188, after careful comparison, we can find all the deleted parameters are unused in the old swagger JSON files. Thus, we remove them in the internal DSMainDev repo.** 

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://msazure.visualstudio.com/One/_workitems/create/Scenario?templateId=d1c215ec-5d6a-4cb3-a193-95ddd0a1ba96&ownerId=85bb92fb-21fe-41ca-9309-47f2c5f57515). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by jianyexi** [comment]
The lintDiff failure , it's because there are  model conflict among the swaggers inside the tag  package-composite-v1 to v4 ,
below is the detail, seems it is adding /removing an item in the enum in the newly added swaggers
```
 '$.definitions.ResourceIdentity.properties.type.enum' has incompatible values (---
- SystemAssigned
, ---
- SystemAssigned
- None
).
```

**Comment by ericshape** [comment]
Hi! @jianyexi 
can you help us to figure out why the PR was failed the javascript SDK automation check? Thanks!
And it seems that java check was failed for the same reason as yesterday.

**Comment by ericshape** [comment]
@akning-ms and @jamestao 
this PR has passed all the checks. Can you help to take a look?

In this PR #11840, I cherry-picked the changes from  #11960 to make it through. I will rebase it once #11960 is merged.

**Comment by ericshape** [comment]
I have rebased this PR. 
The breaking changes are mainly due to removing the unused parameters from the swagger files.

File: specification/sql/resource-manager/readme.azureresourceschema.md [hunk 1]
```diff
@@ -38,20 +39,15 @@ output-folder: $(azureresourceschema-folder)/schemas
 
 # all the input files in this apiVersion
 input-file:
-  - Microsoft.Sql/preview/2020-02-02-preview/managedDatabases.json
-  - Microsoft.Sql/preview/2020-02-02-preview/ServerAzureADOnlyAuthentications.json
-  - Microsoft.Sql/preview/2020-02-02-preview/managedInstances.json
-  - Microsoft.Sql/preview/2020-02-02-preview/importexport.json
-  - Microsoft.Sql/preview/2020-02-02-preview/ManagedInstanceAzureADOnlyAuthentications.json
-  - Microsoft.Sql/preview/2020-02-02-preview/ServerTrustGroups.json
   - ./Microsoft.Sql/preview/2020-02-02-preview/DatabaseSecurityAlertPolicies.json
```
Comment (ericshape) [inline specification/sql/resource-manager/readme.azureresourceschema.md:42]: there is no changes for these files. I just order them as alphabetical and rename the file from abCd to AbCd format

File: specification/sql/resource-manager/readme.md [hunk 2]
```diff
@@ -115,10 +115,9 @@ input-file:
 - Microsoft.Sql/preview/2019-06-01-preview/ServerAzureADAdministrators.json
 - Microsoft.Sql/preview/2019-06-01-preview/syncGroups.json
 - Microsoft.Sql/preview/2019-06-01-preview/syncMembers.json
-- Microsoft.Sql/preview/2019-06-01-preview/FailoverManagedInstance.json
```
Comment (ericshape) [inline specification/sql/resource-manager/readme.md:118]: We just move FailoverManagedInstance to ManagedInstance.json file. Actually, there is no change about API. The operation group name of FailoverManagedInstance is ManagedInstance. Thus, It should be in  ManagedInstance json file.

The reason is that previous developers manually put FailoverManagedInstance API into FailoverManagedInstance.json file. They did not follow the spec file naming rules.

File: specification/sql/resource-manager/readme.md [hunk 3]
```diff
@@ -234,11 +233,9 @@ input-file:
...
-- Microsoft.Sql/preview/2020-02-02-preview/managedDatabases.json
+- Microsoft.Sql/preview/2020-02-02-preview/ManagedDatabases.json
 - Microsoft.Sql/preview/2020-02-02-preview/ServerAzureADOnlyAuthentications.json
-- Microsoft.Sql/preview/2020-02-02-preview/managedInstances.json
-- Microsoft.Sql/preview/2020-02-02-preview/importexport.json
```
Comment (ericshape) [inline specification/sql/resource-manager/readme.md:241]: ImportExport case is similar scenario as FailoverManagedInstance case above.

File: specification/sql/resource-manager/readme.md [hunk 4]
```diff
@@ -716,16 +711,107 @@ These can be regenerated by running the following PowerShell script from this re
...
-- ./Microsoft.Sql/preview/2020-02-02-preview/ManagedServerSecurityAlertPolicies.json
-- ./Microsoft.Sql/preview/2020-02-02-preview/ServerSecurityAlertPolicies.json
-- ./Microsoft.Sql/preview/2020-02-02-preview/operations.json
-- ./Microsoft.Sql/preview/2020-02-02-preview/managedDatabases.json
-- ./Microsoft.Sql/preview/2020-02-02-preview/managedInstances.json
-- ./Microsoft.Sql/preview/2020-02-02-preview/ServerAzureADOnlyAuthentications.json
-- Microsoft.Sql/preview/2020-02-02-preview/importexport.json
-- ./Microsoft.Sql/preview/2020-02-02-preview/ManagedInstanceAzureADOnlyAuthentications.json
-- ./Microsoft.Sql/preview/2020-02-02-preview/ServerTrustGroups.json
+ - ./Microsoft.Sql/preview/2020-02-02-preview/BackupShortTermRetentionPolicies.json
```
Comment (ericshape) [inline specification/sql/resource-manager/readme.md:714]: All the newly added json files are carried from previous versions. They are NOT the new introduced API endpoints.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2020-02-02-preview/ServerTrustGroups.json [hunk 5]
```diff
@@ -240,16 +240,16 @@
...
           {
-            "$ref": "#/parameters/ResourceGroupParameter"
+            "$ref": "../../../common/v1/types.json#/parameters/ResourceGroupParameter"
           },
           {
             "$ref": "#/parameters/ManagedInstanceNameParameter"
           },
           {
-            "$ref": "#/parameters/SubscriptionIdParameter"
+            "$ref": "../../../common/v1/types.json#/parameters/SubscriptionIdParameter"
```
Comment (ericshape) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2020-02-02-preview/ServerTrustGroups.json:249]: In these serverXYZ json files, we only change the common type parameters/definitions to reference and remove previous local inline parameters/definitions.

**Comment by ericshape** [comment]
the failures of JS and Ruby SDK are not due to our changes. And APIDocPreview and cross-version breaking changes have been stuck over 12 hours.

**Review by jamestao — APPROVED** [review]

**Comment by akning-ms** [comment]
comments:
1. "autoRotationEnabled" is existing in 2020-08-01-preview, but not in 2020-02-02-preview. it is correct?  in ManagedInstanceEncryptionProtectors.json
2. Should DevOpsAudit.json be changed to name ServerDevOpsAudit.json as 2020-08-01. looks there is big different between them
3. DatabaseExtension is in 2020-08-01, it is correct?

Recommendation:
As it is huge PR. compare is not easy. so I have to compare with 2020-08-01-preview.
In the future. suggest to move origin files from all versions without any change to commit a base version. the commit actually change. it will make review much easy in github.
BTW, in your next new version, you can use below URL to copy an existing version(e.g 2020-08-01-preview) to a new version:
https://portal.azure-devex-tools.com/branch/create
Tools will help you create folder and change version # and commit a base version. it make new version much easy.

**Review by bcham — APPROVED** [review]

**Comment by ericshape** [comment]
@akning-ms 
I have addressed your comments via MSFT internal email. Thanks!

**Pull request merged**